pgwire/src/main/java/io/crate/protocols/postgres/parser/PgArrayASTVisitor.java

```diff
@@ -50,7 +50,27 @@ public Object visitArray(PgArrayParser.ArrayContext ctx) {
 
     @Override
     public Object visitValue(PgArrayParser.ValueContext ctx) {
-        return convert.apply(processArrayItem(ctx.getText().getBytes(UTF_8)));
+        PgArrayParser.StringContext stringContext = ctx.string();
+        if (stringContext == null) {
+            return convert.apply(processArrayItem(ctx.getText().getBytes(UTF_8)));
+        }
+        Object value = visit(stringContext);
+        if (value == null) {
```

Comment:
Made the quoted/non-quoted distinction to be able to handle `null` without quotes as real null value instead of a `null` string literal. Not sure if there would be a better way to accomplish this.

Reply:
can we drop https://github.com/crate/crate/pull/9508/files#diff-56e21a74f1c628321abf2180bc314273R90-R92 such as we have a dedicated rules to match quoted and unquoted identifiers? otherwise, lgtm

Reply:
@kovrus pushed a fixup